core/integration/gpu_test.go

```diff
@@ -0,0 +1,187 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// imageName defines the name of Nvidia's CUDA contaimer image:
+	imageName = "nvidia/cuda"
+	// imageFlavor sets the default image flavor, as defined in: https://hub.docker.com/r/nvidia/cuda
+	imageFlavor = "base"
+
+	// torchTensorsSample is a PyTorch sample taken from the official docs:
+	// https://pytorch.org/tutorials/beginner/pytorch_with_examples.html#pytorch-tensors
+	torchTensorsSample = `# -*- coding: utf-8 -*-
+import torch
+import math
+dtype = torch.float
+device = torch.device("cuda:0")
+# Create random input and output data
+x = torch.linspace(-math.pi, math.pi, 2000, device=device, dtype=dtype)
+y = torch.sin(x)
+# Randomly initialize weights
+a = torch.randn((), device=device, dtype=dtype)
+b = torch.randn((), device=device, dtype=dtype)
+c = torch.randn((), device=device, dtype=dtype)
+d = torch.randn((), device=device, dtype=dtype)
+learning_rate = 1e-6
+for t in range(2000):
+    # Forward pass: compute predicted y
+    y_pred = a + b * x + c * x ** 2 + d * x ** 3
+    # Compute and print loss
+    loss = (y_pred - y).pow(2).sum().item()
+    if t % 100 == 99:
+        print(t, loss)
+    # Backprop to compute gradients of a, b, c, d with respect to loss
+    grad_y_pred = 2.0 * (y_pred - y)
+    grad_a = grad_y_pred.sum()
+    grad_b = (grad_y_pred * x).sum()
+    grad_c = (grad_y_pred * x ** 2).sum()
+    grad_d = (grad_y_pred * x ** 3).sum()
+    # Update weights using gradient descent
+    a -= learning_rate * grad_a
+    b -= learning_rate * grad_b
+    c -= learning_rate * grad_c
+    d -= learning_rate * grad_d
+print(f'Result: y = {a.item()} + {b.item()} x + {c.item()} x^2 + {d.item()} x^3')
+	`
+	gpuTestsEnabledEnvName = "DAGGER_GPU_TESTS_ENABLED"
```

Comment:
Is there any reason @matiasinsaurralde why we are not re-using the same environment variable for this, `_EXPERIMENTAL_DAGGER_GPU_SUPPORT ` ?

Reply:
I think we target `_EXPERIMENTAL_DAGGER_GPU_SUPPORT` for the actual users and `DAGGER_GPU_TESTS_ENABLED` for our CI but we could potentially make it the same.

Reply:
We would set `DAGGER_GPU_TESTS_ENABLED` whenever CI is ready to run the tests on a GPU enabled instance.

Reply:
OK, in which case let's leave this as is for now.

I can imagine us wanting to run all tests against a GPU enabled Engine so that we ensure it behaves the same as the non-GPU one.

To take the above further, if both Engine images will use the same base image - e.g. Ubuntu - then this will not be necessary.

As this can go in a few directions, happy to defer this decision when we know more.

No further action required, all good @matiasinsaurralde!